src/buf/buf_impl.rs

```diff
@@ -285,7 +286,7 @@ pub trait Buf {
     ///
     /// This function panics if there is no more remaining data in `self`.
     fn get_u8(&mut self) -> u8 {
-        assert!(self.remaining() >= 1);
+        assert!(self.chunk().len() >= 1);
```

Comment:
I mean, the index bounds check happens just after this, I must assume these asserts are mostly to give a better panic message. Perhaps we could just make them debug_asserts?

@carllerche you added these asserts originally, I think?